deployment/deploy/deploy.sh

```diff
@@ -104,13 +104,46 @@ else
     UPDATE_NGINX=1
 fi
 
+filterConf() {
+    SRC="$1"
+    DST="$2"
+    SERVER_NAME="$3"
+    PROXY_REDIRECT="$4"
+    INTERNAL_PORT="$5"
+
+    if [ ! -f "$DST" ]; then
+        sed -e "s/\${DOMAIN_NAME}/${DOMAIN_NAME}/" "$SRC" | \
+        sed -e "s/\${SERVER_NAME}/${SERVER_NAME}/" | \
+        sed -e "s/\${PROXY_REDIRECT}/${PROXY_REDIRECT}/" | \
+        sed -e "s/\${INTERNAL_PORT}/${INTERNAL_PORT}/" >"$DST"
```

Comment:
Does each variable occur only once in the config?  Otherwise a `/g` flag would be missing for sed.  It might not hurt to add it anyway to avoid later surprises (although I don't know the structure of that config file specifically...).

Reply:
Looking at the site template further below, I see the port being used twice for example.  So how did that work, or maybe I am missing something?  Is `/g` the default, and I was deceived my whole live?  Need to check...

Reply:
Learn something new every day: `/g` only changes behavior for multiple matches *on the same line*, not like global globally:
```sh
$ export PORT=123
$ echo 'FOO${PORT}BAR${PORT}BUZ' | sed -e "s/\${PORT}/$PORT/g"
FOO123BAR123BUZ

$ printf  'FOO${PORT}BAR${PORT}BUZ\n' | sed -e "s/\${PORT}/$PORT/"
FOO123BAR${PORT}BUZ

$ printf  'FOO${PORT}\nBAR${PORT}BUZ\n' | sed -e "s/\${PORT}/$PORT/"
FOO123
BAR123BUZ
```

Anyway, sorry for the rambling, I leave it to you to judge if that's relevant :-)

Reply:
I read your comments one at a time. For the first, I started looking at man pages (flags for s or /g are not even mentioned in my sed manual). Then online, where g is mentioned sometime. But I knew that it worked, so I was trying to figure out why. I suspected sed worked on one line at a time, but didn't find that clearly specified.

Then I read the second comment, and I was going to reply with
```
echo -e "AbcdAbcd\nAbcdAbcd"|sed 's/A/X/'
```
which should clarify things.

And then I saw your third comment.
The lesson here is that I should probably read all comments first.

Reply:
And, yes, /g should be there.

Reply:
Done

Reply:
> The lesson here is that I should probably read all comments first.
 Or maybe I should think things through before I comment... :-P  Thanks for indulging me!

Reply:
I think my point was that I went through a somewhat similar thought process and, as such, reading first would have saved me some time.